### DISCUSSION TITLE
Setting up Laravel Octane for Production – Seeking Guidance and Best Practices

### DISCUSSION THREAD

**Discussion by rupeshstha** [laravel/framework#48785] [Q&A]
Hello fellow developers!

I'm currently in the process of setting up Laravel Octane for a production environment and I'd love to hear from those of you who have experience with it. Octane promises significant performance improvements, and I want to make sure I'm following best practices for a seamless deployment.

If you have successfully deployed Laravel Octane in production, could you kindly share your insights and guidance on the following:

**Configuration Best Practices**: What are the recommended configuration settings and optimizations for Octane in a production environment?

**Security Considerations**: What security measures and best practices should be taken into account when using Octane in production?

**Experiences and Pitfalls**: Have you encountered any challenges or unexpected issues during your Octane journey in a production environment? How did you overcome them?

Please feel free to share any additional insights, tips, or resources that you think would be valuable to someone setting up Laravel Octane for production. Your experiences and expertise will be greatly appreciated!

Let's build a knowledge base on this topic to help others who are embarking on their own Octane adventures. Thank you in advance for your contributions!

**Comment by nilskretschmer** [comment]
Hey @rupeshstha 

This is of course a very general topic, because production environments vary a lot.

But I think the Laravel Docs already explain the basics to get you started. If I were to make a recommendation, it would be something like this:

1. Use Docker Images for your App (and related services like Redis etc.)
2. Create your own Dockerfile's and build your production images (use php-fpm as a base for your Laravel App, maybe with Alpine Linux image version if you need to get small image sizes.)
- Make sure to include the `Swoole` PECL extension and other extensions used by your Laravel App during build
- Stick to the Laravel Deployment Guide and Octane Docs for specific Deployment Requirements as these may change
3. Use nginx as a reverse proxy and use proxy_pass to pass incoming requests to the octane server
4. Use Docker Swarm/Kubernetes for Scalable Production Deployments
5. Use something like Terraform for configuration/state management
6. Make sure that on each new rollout all services/docker containers are restarted - including your octane server - so that new changes in your code will be reflected

But as your question being general this is just a very general approach and this needs to be adjusted depending on your actual production environment.
